navit/navit.c

```diff
@@ -2459,6 +2459,27 @@ void navit_set_center_cursor(struct navit *this_, int autozoom, int keep_orienta
         navit_autozoom(this_, &nv->coord, nv->speed, 0);
 }
 
+/**
+ * @brief Drags (moves) the map
+ *
+ * Drags (moves) the map from origin point to the destination point
+ *
+ * @param navit The navit instance
+ * @param origin The point point where the drag starts
+ * @param destination The point where to map should be dragged to
+ * @returns nothing
+ */
+
+void navit_drag_map(struct navit *this_, struct point *origin, struct point *destination) {
+    update_transformation(this_->trans, origin, destination);
```

Comment:
Since this shall get a "public" function, I suggest to do some sanity checking here, like aborting if some of the parameters is NULL, issuing some err message if this is not expected. May help future users of this in calling the function right.